### ISSUE TITLE
Performance on the MVTec dataset

### ISSUE THREAD

**Issue by cuicathy** [iolag/UPD_study#6]
Hello, 

Have you evaluated these methods using the MVTec dataset? I attempted to use your code on the MVTec dataset, but the outcomes were significantly inferior compared to those documented in the original paper. Specifically, the AUC for most methods was about 0.5-0.8 after 2000 training steps, except the method RD and DFR. 

I used the following command. I changed the small testing set in your code to the complete testing set and observed the anom_val/sample-auroc in the wandb. 
UPD_study/models/H-TAE-S/HTAEStrainer.py --shuffle t -ev t -mod MyData --percentage 100 --eval False  --seed 10 --val_frequency 200 --log_frequency 200  --anom_val_frequency 200

Is there any problem with my usage? Look forward to your reply. 

Thank you in advance!

**Comment by iolag** [comment]
Hey,

We did not evaluate on MVTec. As most methods did not work well out of the box, as they were optimized for the data they were developed for, we tried to do some optimization ourselves especially with methods that completely failed on the datasets we used.  I especially remember the case of CFLOW-AD, which completely failed on MRIs even though it was one of the best performers on MVTec at the time. In that case, I had to change the Resnet layers that are used for feature extraction in order to get decent performance. This for sure had a negative effect on its MVTec performance. 

On the other hand, RD worked really great with the original hyperparameters and we kept it largely unchanged, which explains the results you get. I'm not sure why PADIM isn't performing great for you as we used the original implementation, if I remember correctly.  

As for the rest of the methods, I would not expect their performance to translate to MVTec as they were optimized on very different types of data.

**Comment by cuicathy** [comment]
Hi iolag, 

Thanks very much for your response. Do you remember what parameter did you tune? And how did you tune them (e.g., grid search)? 

About the PADIM, I did not run it before last week since I only compared the models that required training for epochs last time. For the PADIM, it performed about AUC 0.96 (anomaly detection) on the hazelnut class of the MVTec.

**Comment by iolag** [comment]
Hi,

sorry for the late response.

For VAE, I remember using grid search on a lot of hyper-parameters. On others, I empirically changed certain parameters,  mostly based on things I noticed during early experiments. I go into a bit more detail on which ones I changed compared to the official implementation in the end of my [Masters thesis](https://mega.nz/file/87YlSQwB#Z6pnlMdQqnyfgt7a5vpD844o-Wm0rdSbnR4gv7hAx1M). There, only methods that I used before the journal revisions are mentioned, but the ones missing I mostly implemented unchanged.
